src/modules/IscsiClientLib.rb

```diff
@@ -775,29 +775,40 @@ def getCurrentNodeValues
       nodeInfoToMap(ret["stdout"] || "")
     end
 
+    # Checks whether iSCSI session (values provided as hash) is iBFT session
+    #
+    # @param  [Hash]      iSCSI node values as hash
+    # @return [Bool]      is iSCSI session booted from firmware?
+    #
+    def isiBFT?(node_info)
+      ret = false
+      ibft = getiBFT
+
+      if (!ibft.empty?)
```

Comment:
Here you can merge all conditions and can get rid of the `ret` variable:

```ruby
return !ibft.empty? && ibft["iface.transport_name"] == node_info["iface.transport_name"] &&
  ibft["iface.initiatorname"] == node_info["iface.initiatorname"] &&
  ibft["node.name"] == node_info["node.name"] &&
  ibft["node.conn[0].address"] == node_info["node.conn[0].address"]
```

And I'd even refactor the compare functionality (repeating all the keys is error prone):

```ruby
def equal_nodes?(n1, n2)
  return false if n1.empty?
  
  # FIXME: maybe define a constant?
  keys = [
    "iface.transport_name",
    "iface.initiatorname",
    "node.name",
    "node.conn[0].address"
  ]
  
  keys.all? { |key| n1[key] == n2[key] }
end
```

and then in the `isiBFT?` method simply call:
```ruby
  return equal_nodes?(getiBFT, node_info)
```